packages/callback/android/src/main/java/io/flutter/plugins/callback/CallbackPlugin.java

```diff
@@ -0,0 +1,47 @@
+package io.flutter.plugins.callback;
+
+import java.util.HashMap;
+import java.util.Map;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/**
+ * Android implementation of the CallbackPlugin.
+ */
+public class CallbackPlugin implements MethodCallHandler {
+  /**
+   * Plugin registration.
+   */
+  public static void registerWith(Registrar registrar) {
+    CallbackPlugin plugin = new CallbackPlugin();
+    final MethodChannel channel = new MethodChannel(registrar.messenger(), "callback");
+    channel.setMethodCallHandler(plugin);
+    registrar.publish(plugin);
+  }
+
+  private final Map<String, Runnable> callbacks = new HashMap<>();
+
+  public void registerCallback(String id, Runnable callback) {
+    callbacks.put(id, callback);
+  }
```

Comment:
I may have misunderstood something. But don't you want a list of callbacks per id here? If there is only one, the callback plugin seems to be just a wrapper around a `MethodChannel`, with `Runnable` instead of `MethodCallHandler`.

See comments starting with "Alternatively, [...]" below.